discovery.rst

```diff
@@ -2,12 +2,16 @@ Discovery
 =========
 
 The discovery service allows to find and use installed resources.
+
 Under the hood it uses `Puli`_ for the discovery logic. All of our packages provide Puli resources.
 Discovery is simply a convenience wrapper to statically access clients and factories for when
 Dependency Injection is not an option. Discovery is useful in libraries that want to offer
 zero-configuration services relying on the virtual packages. If you have Dependency Injection available,
 using Puli directly is more elegant (see for example the Symfony HttplugBundle).
 
+Consumers of libraries using discovery still need to make sure they install one of the implementations.
+Discovery can only find installed code, not fetch code from other sources.
```

Comment:
There is many question about discovery, writing more things about (even if its redundant) is for the best IMO